compiler/src/dotty/tools/dotc/transform/localopt/StringInterpolatorOpt.scala

```diff
@@ -32,15 +32,29 @@ class StringInterpolatorOpt extends MiniPhase {
     }
   }
 
+  private object StringContextIdent {
+    def unapply(tree: Ident)(implicit ctx: Context): Option[Ident] = {
+      if (tree.symbol.eq(defn.StringContextModule)) Some(tree) else None
+    }
+  }
+
+  private object StringContextApply {
```

Comment:
I see that this extractor is just a test and does not return any result (but the tree itself). You can return a `Boolean` instead:
```scala
private object StringContextApply {
  def unapply(tree: Select)(implicit ctx: Context): Boolean =
    tree.symbol.eq(defn.StringContextModule_apply) && {
      val qualifier = tree.qualifier
      qualifier.isInstanceOf[Ident] && qualifier.symbol.eq(efn.StringContextModule)
    }
}
```